Gems/Terrain/Assets/Shaders/Terrain/TerrainCommon.azsli

```diff
@@ -104,9 +110,23 @@ void CalculateTerrainPosition(
     }
 
     float heightScale = worldData.m_max.z - worldData.m_min.z;
-    float height = worldData.m_min.z + input.m_zposition * heightScale;
+    float height = worldData.m_min.z + input.m_zPosition * heightScale;
+
     worldPosition = float3(GetWorldXYPosition(patchData, input.m_xyPosition), height);
     cullDistance = TerrainSrg::CalculateCullDistance(patchData.m_lodLevel, worldPosition);
+    normal = float3(input.m_xyNormal, 0.0);
+
+    if (o_useTerrainClod)
+    {
+        float quadSize = patchData.m_xyScale * rcp(64.0);
+        float clodPadding = quadSize * sqrt(2.0) * 2;
```

Comment:
Some named constants or comments explaining the number choices here would be helpful!